cpp/include/raft/spatial/knn/detail/ann_utils.cuh

```diff
@@ -192,14 +191,18 @@ __global__ void argmin_along_rows_kernel(uint32_t n_rows,
  * NB: device-only function
  * TODO: specialize select_k for the case of `k == 1` and use that one instead.
  *
+ * @tparam IdxT   index type
+ * @tparam LabelT label type
+ *
  * @param n_rows
  * @param n_cols
  * @param[in] a device pointer to the row-major matrix [n_rows, n_cols]
  * @param[out] out device pointer to the vector of selected indices [n_rows]
  * @param stream
  */
+template <typename IdxT, typename LabelT>
 inline void argmin_along_rows(
```

Comment:
I've noted there in the docs that it would be nice to merge this into select-top-k as a special case https://github.com/rapidsai/raft/blob/2e98138c7d022e754e2b2f3bbab377d9d47150c3/cpp/include/raft/spatial/knn/detail/ann_utils.cuh#L193

I suggest I will do this as soon as ivf-pq implementation is merged. Here's the linked issue https://github.com/rapidsai/raft/issues/853

Reply:
Argmin and argmax themselves are useful primitives. We do have an argmax primitive that uses cub underneath currently. We could create an argmin from that, or use this one. Either way, i think it's generally useful and could be moved to `raft::matrix::argmin`